### ISSUE TITLE
Adding file extensions that are already in use

### ISSUE THREAD

**Issue by detzt** [danielscherzer/GLSL#11]
### Installed product versions
- Visual Studio: [2017 Enterprise]
- This extension: [example 0.2.33]

### Description
Adding .vsh or .fsh file extentions in the options leads to errors upon opening a project.
(Adding other file extensions like .tcsh or .tesh works just fine).
ActivityLog.xml says the file extension is registered twice. Unfortunately I was not able to find out where they were already registered.
Clean install of VS2017 with C++ and Graphics tools, but no further plugins.

### Steps to recreate
1. Install GLSL plugin
2. Add .vsh or .fsh to the file extensions
3. Open a project containing such files

### Current behavior
Throws errors upon loading the project and does not highlight the files with the 

### Expected behavior
Open the project and correctly display the files (highlighting, completion, etc.)

**Comment by detzt** [comment]
Thank you for looking into this that quickly!
I have no experience with developing VS extensions, but there is an entry in the options to configure which editor should be used for which file type. See https://msdn.microsoft.com/en-us/library/4k7w5e5s.aspx  
If I remember correctly, NShader used this to configure the file extensions, but it's bin a while since I configured that plugin under VS2015.
Anyways, the GLSL plugin does not appear in the list of available editors. Maybe this works better than stealing extensions? :smirk:

**Comment by nshockwa** [comment]
<img width="313" alt="capture" src="https://user-images.githubusercontent.com/12228840/46325649-24f7c880-c5ae-11e8-884e-b966ac6ad3bd.PNG">

Any way around this?
I can't find where it's registered either :/

**Comment by danielscherzer** [comment]
This message means that probably some other vs extension has registered a language integration for the extension `.glsl`, for instance NShade and some Unity tools. I found no graceful way of two vs extensions handling the same file extension. One way is to use another file extension for the shaders you use with my vs extension, like .vert, .frag ... This is configurable from the options page of my vs extension. Another way is to disable all vs extensions except mine. Check if glsl is now working with mine and then switching on each extension in turn to find the "guilty" one. Then maybe you can configure the "guilty" one to not use glsl. I would be interested to know the name of the extension, so I can note incompatibilities in the documentation and contact the makers about the problem. Cheers!

**Comment by danielscherzer** [comment]
> 
> 
> Thank you for looking into this that quickly!
> I have no experience with developing VS extensions, but there is an entry in the options to configure which editor should be used for which file type. See https://msdn.microsoft.com/en-us/library/4k7w5e5s.aspx
> If I remember correctly, NShader used this to configure the file extensions, but it's bin a while since I configured that plugin under VS2015.
> Anyways, the GLSL plugin does not appear in the list of available editors. Maybe this works better than stealing extensions? 😏

Yep this could work! Do you have any idea of how to register an extension as an "Extension and Editing Experience"?

**Comment by nshockwa** [comment]
Tried uninstalling unity tools for visual studio in addition to disabling all my extensions to no avail. However, after uninstalling VS and it's program files folder then reinstalling, it works!

**Comment by danielscherzer** [comment]
Thank you for your effort! Maybe it is VS that is to blame. Recently a lot of bugs are in these almost weakly releases that vanish a few update cycles later. So i hope this problem is settled.

**Comment by Xottab-DUTY** [comment]
When this silly window appears VS stops any work and waits when user closes the window. That's awful. When VisualAssistX shows it's daily tips window it allows VS to work, but that `Extension .xxxx is ignored..` window doesn't! Could you fix that or at least show a notification instead of a window?

**Comment by danielscherzer** [comment]
Hopefully fixed!
